sdk/src/androidTest/java/com/mapbox/maps/ViewAnnotationTest.kt

```diff
@@ -0,0 +1,729 @@
+        resId = R.layout.view_annotation,
+        options = viewAnnotationOptions {
+          geometry(CAMERA_CENTER)
+          anchor(ViewAnnotationAnchor.TOP_LEFT)
+          allowOverlap(true)
+        }
+      )
+      mainHandler.postDelayed(
+        {
+          assertTrue(mapView.hasChildView(firstView))
+          assertEquals(mapboxMap.pixelForCoordinate(SHIFTED_CENTER).x, firstView.translationX.toDouble(), ADMISSIBLE_ERROR_PX)
+          assertEquals(mapboxMap.pixelForCoordinate(SHIFTED_CENTER).y, firstView.translationY.toDouble(), ADMISSIBLE_ERROR_PX)
+          assertTrue(mapView.hasChildView(secondView))
+          assertEquals(mapboxMap.pixelForCoordinate(CAMERA_CENTER).x, secondView.translationX.toDouble(), ADMISSIBLE_ERROR_PX)
+          assertEquals(mapboxMap.pixelForCoordinate(CAMERA_CENTER).y, secondView.translationY.toDouble(), ADMISSIBLE_ERROR_PX)
+          MatcherAssert.assertThat(
+            mapView.getChildViewIndex(secondView),
+            Matchers.greaterThan(mapView.getChildViewIndex(firstView))
+          )
+          latch.countDown()
+        },
+        VIEW_PLACEMENT_DELAY_MS
+      )
+    }
+    if (!latch.await(CONFIGURE_TIMEOUT_MS, TimeUnit.MILLISECONDS)) {
+      throw TimeoutException()
+    }
+  }
+
+  @Test
+  fun addTwoViewAnnotationsAllowOverlapFalse() {
+    val latch = CountDownLatch(1)
+    mainHandler.post {
+      val firstView = viewAnnotationManager.addViewAnnotation(
+        resId = R.layout.view_annotation,
+        options = viewAnnotationOptions {
+          geometry(SHIFTED_CENTER)
+          anchor(ViewAnnotationAnchor.TOP_LEFT)
+          allowOverlap(false)
+        }
+      )
+      val secondView = viewAnnotationManager.addViewAnnotation(
+        resId = R.layout.view_annotation,
+        options = viewAnnotationOptions {
+          geometry(CAMERA_CENTER)
+          anchor(ViewAnnotationAnchor.TOP_LEFT)
+          allowOverlap(false)
+        }
+      )
+      mainHandler.postDelayed(
+        {
+          assertFalse(mapView.hasChildView(firstView))
+          assertTrue(mapView.hasChildView(secondView))
+          assertEquals(mapboxMap.pixelForCoordinate(CAMERA_CENTER).x, secondView.translationX.toDouble(), ADMISSIBLE_ERROR_PX)
+          assertEquals(mapboxMap.pixelForCoordinate(CAMERA_CENTER).y, secondView.translationY.toDouble(), ADMISSIBLE_ERROR_PX)
+          latch.countDown()
+        },
+        VIEW_PLACEMENT_DELAY_MS
+      )
+    }
+    if (!latch.await(CONFIGURE_TIMEOUT_MS, TimeUnit.MILLISECONDS)) {
+      throw TimeoutException()
+    }
+  }
+
+  @Test
+  fun addTwoViewAnnotationsOneSelectedAllowOverlapTrue() {
+    val latch = CountDownLatch(1)
+    mainHandler.post {
+      val firstView = viewAnnotationManager.addViewAnnotation(
+        resId = R.layout.view_annotation,
+        options = viewAnnotationOptions {
+          geometry(SHIFTED_CENTER)
+          anchor(ViewAnnotationAnchor.TOP_LEFT)
+          allowOverlap(true)
```

Comment:
What will happen if one `allowOverlap` and another doesn't?

Reply:
added two tests, behaviour looks logical for me, check them out please:
```
addTwoViewAnnotationsOneAllowOverlapTrueAnotherFalse
addTwoViewAnnotationsOneAllowOverlapFalseAnotherTrue
```